include/awkward/kernel-dispatch.h

```diff
@@ -130,6 +130,12 @@ namespace awkward {
         void operator()(T const *ptr) { }
     };
 
+    /// @brief Returns the number of the device associated with the pointer
+    const int32_t
+    lib_device_num(
+      kernel::lib ptr_lib,
+      void* ptr);
+
```

Comment:
Almost all of the scalar integers in Awkward Array are `int64_t`. (NumpyArray has `ssize_t` to match what comes out of pybind11, but now I think that was a mistake: it would have been cleaner to convert them once in the pybind11 interface than everywhere in the interfaces between NumpyArray and everything else.)

Perhaps the device number returned by CUDA is 32-bit, but is there any reason not to immediately convert it to 64-bit so that it can be passed around as 64-bit without compiler warnings? (Some compiler warnings are raised by just one or two of the compilers in CI: gcc on Linux, cling on MacOS, and msvc on Windows.)